### PR TITLE
Update code for API changes in thclient 1.6 (#610)

### PR DESCRIPTION
**Pull request by whimboo**
this PR will fix issue #610. @edmorley, you wanted to review? :) Thanks

### CODE REVIEW

**Comment by whimboo** [comment]
To be safer in the future I have also updated the dependencies to be strict. Otherwise API changes can cause bustage.

File: jenkins-master/jobs/scripts/workspace/treeherder.py [hunk 1]
```diff
@@ -145,22 +146,25 @@ def retrieve_revision_hash(self):
...
         job.add_submit_timestamp(int(time.time()))
 
-        job_collection = TreeherderJobCollection(job_type='update')
+        job_collection = TreeherderJobCollection()
         job_collection.add(job)
 
         # self.logger.info
         print('Sending results to Treeherder: %s' % job_collection.to_json())
 
+        auth = TreeherderAuth(self.key, self.secret, user=self.project)
```
Comment (edmorley) [inline jenkins-master/jobs/scripts/workspace/treeherder.py]: `user` isn't a keyword argument, so I believe this should be:
```
auth = TreeherderAuth(self.key, self.secret, self.project)
```

**Comment by edmorley** [comment]
Looks good to me apart from the small change above - sorry for the hassle with the backwards incompatibilities! :-)

File: jenkins-master/jobs/scripts/workspace/treeherder.py (hunk 1 above)
Comment (whimboo) [inline jenkins-master/jobs/scripts/workspace/treeherder.py]: It's not but it would work. But yes, for consistency I will remove it.

**Comment by whimboo** [comment]
No worries and thanks for the review. I pushed the fix and will merge once travis is happy.

**Pull request merged**